RevenueCatUI/CustomerCenter/ViewModels/SubscriptionDetailViewModel.swift

```diff
@@ -69,14 +69,32 @@ final class SubscriptionDetailViewModel: BaseManageSubscriptionViewModel {
             )
         }
 
+    func didAppear() {
+        actionWrapper.onCustomerCenterPromotionalOfferSuccess { [weak self] in
+            self?.refreshPurchase()
+        }
+
+        actionWrapper.onCustomerCenterShowingManageSubscriptions { [weak self] in
+            self?.customerInfoViewModel.manageSubscriptionsSheet = true
+        }
+
+        customerInfoViewModel.$manageSubscriptionsSheet
+            .filter { !$0 }
+            .sink { [weak self] _ in
+                self?.refreshPurchase()
+            }
```

Comment:
maybe confusing use of `filter` here. Why not just this?
```suggestion
            .sink { [weak self] _ in
                if !$0 {
                    self?.refreshPurchase()
                }
            }
```